api/src/test/java/org/apache/iceberg/expressions/TestMiscLiteralConversions.java

```diff
@@ -231,7 +231,6 @@ public void testInvalidDecimalConversions() {
         Types.TimeType.get(),
         Types.TimestampType.withZone(),
         Types.TimestampType.withoutZone(),
-        Types.DecimalType.of(9, 4),
```

Comment:
Why was this removed?

Reply:
Because it's no longer an invalid conversion to call `DecimalLiteral.to()` on a decimal which has a scale of 2 to convert to another decimal which has a scale of 4

Reply:
Right, it is no longer invalid.